### ISSUE TITLE
Indendation of low-precedence  infix operators

### ISSUE THREAD

**Issue by AshesITR** [r-lib/styler#1150]
Related to: https://github.com/r-lib/lintr/issues/1718

```r
y ~
  x1 +
    x2
```

Has parse tree `~(y, +(x1, x2))` so the `~` ends in line 3 and should thus cause an indentation for lines 2 and 3.
`+` ends in line 3 and should cause _another_ level of indentation on line 3.

``` r
styler::style_text("
y ~
  x1 +
    x2
")
#> y ~
#>   x1 +
#>   x2
```

<sup>Created on 2023-09-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
The style guide doesn't really say much about indentation. Perhaps we could raise that; meanwhile, editing code that _doesn't_ violate anything in the style guide generally goes against styler's intention, right?

**Comment by AshesITR** [comment]
Note that `a + b + c` is `(a + b) + c` whereas `a ~ b + c` is `a ~ (b + c)`.
Therefor, indentation shouldn't be the same for the two.

``` r
styler::style_text("
y +
  x1 +
  x2
")
#> y +
#>   x1 +
#>   x2
```

<sup>Created on 2023-09-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by HaHeho** [comment]
I can add the following similar example that yields malformed code (the input is the correctly formatted code that I would expect).
```R
> styler::style_text("
+ stats::model.matrix(
+   ~ a
+   + b
+   + c,
+   dd
+ )
+ ")
stats::model.matrix(
  ~ a
  + b
    + c,
  dd
)
```
Note that considerations around the operator precedence do not apply here since they don't actually function as mathematical operators in this context. I believe something similar may arise around the `ggplot2::facet_grid` function.

**Comment by lorenzwalthert** [comment]
> Note that considerations around the operator precedence do not apply here since they don't actually function as mathematical operators in this context

I don’t think we can differentiate these cases here in styler. Also, the issue caused some headache already in the past and I can’t recall the reasons for tie current implementation and if your representative was intentional or not (but it could probably be for d searching the issues) and given that the style guide is silent on it and the syntax is generally not used very often, this has low priority for me.
